src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -223,20 +228,20 @@ private void InterceptCallAndAdjustArguments(
                 },
                 extraArgument: attributeLocation);
 
-            if (!MemberSignatureComparer.InterceptorsStrictComparer.Equals(method, symbolForCompare))
+            if (!MemberSignatureComparer.InterceptorsStrictComparer.Equals(method, interceptorForCompare))
             {
                 this._diagnostics.Add(ErrorCode.WRN_InterceptorSignatureMismatch, attributeLocation, method, interceptor);
             }
 
-            method.TryGetThisParameter(out var methodThisParameter);
-            var interceptorThisParameterForCompare = needToReduce ? interceptor.Parameters[0] :
-                interceptor.TryGetThisParameter(out var interceptorThisParameter) ? interceptorThisParameter : null;
-            switch (methodThisParameter, interceptorThisParameterForCompare)
+            ParameterSymbol? methodReceiverParameter = getReceiverParameter(method, classicExtensionNotInvokedAsStatic: invokedAsExtensionMethod);
```

Comment:
>invokedAsExtensionMethod [](http://example.com/codeflow?start=120&length=24)

This value feels wrong. ```methodThisParameter``` didn't depend on ```invokedAsExtensionMethod``` before and the renamed value ```methodReceiverParameter``` shouldn't depend on it too, meaning we should always pass ```false``` here. Otherwise, it looks like there is a behavior change here for a legacy scenario, however, there is no indication (in a comment or in a PR description) that there is an intent to fix a bug.